libqtile/core/manager.py

```diff
@@ -361,6 +361,35 @@ def _process_screens(self) -> None:
             )
             self.screens.append(scr)
 
+    def reconfigure_screens(self, ev=None):
+        """
+        This can be used to set up screens again during run time. Intended usage is to
+        be called when the screen_change hook is fired, responding to changes in
+        physical monitor setup by configuring qtile.screens accordingly. The ev kwarg is
+        ignored; it is here in case this function is hooked directly to screen_change.
+        """
+        screens = []
+        screen_info = self.core.get_screen_info()
+        logger.info("Reconfiguring screens to: {}".format(screen_info))
+
+        for i, (x, y, w, h) in enumerate(screen_info):
+            if i < len(self.screens):
+                scr = self.screens[i]
+            elif i < len(self.config.screens):
+                scr = self.config.screens[i]
+            else:
+                scr = Screen()
+
+            scr._configure(self, i, x, y, w, h, self.groups[i])
+            scr.group.layout_all()
+            screens.append(scr)
+
+        for group in self.groups:
+            if group.screen and group.screen not in screens:
+                group.screen = None
```

Comment:
Can you tell me more about this? In particular, why are we not assigning the leftovers to one of the new screens?

Reply:
This is more for when there are now fewer screens present: any groups that were previously on a screen that is now not there will have have their `screen` reset to `None`. When there are more screens available, then nothing will happen because `group.screen` will be in `screens`.

That does make me wonder -- what happens (on master) when there are more screens than there are groups? I have not accounted for that possibility in this PR.

Reply:
Yeah, my question was badly formulated but that was my concern as well.

Also, I'm thinking that we probably hold a lot more references to the removed screens than just in groups. What should we do with them?

Reply:
I just checked all references to screens the other things that save screens directly (rather than accessing via a group or `qtile`) are: `Gap`s/`Bar`s, `widget.Systray`, and `window.Static`. I guess that sounds about right, too.

- Static windows can just be re-positioned to be on the first screen (TODO).
- Gaps and Bars do not need to be handled further; they exist within `qtile.config.screens` and are simply reconfigured when their screen (index) is reconfigured, and this does very little as bars are already set up for multiple calls to `Bar._configure`. Widgets on bars from dormat screens continue to live (and poll) but cannot be accessed until their screen is restored.
- `widget.Systray` _may_ show possibly unexpected behaviour, but I'm not familiar enough with X's systrays to know for sure. I do however use this widget and PR together (and have done for months) and have had no problem. The widget saves an atom telling the server which screen it is on. The screen number doesn't really change (it just disappears) so it will be correct when the screen is restored.